client_cli/src/main.rs

```diff
@@ -20,7 +20,21 @@ use iroha_crypto::prelude::*;
 use iroha_data_model::prelude::*;
 use iroha_primitives::addr::SocketAddr;
 
-/// Metadata wrapper, which can be captured from cli arguments (from user supplied file).
+/// A value wrapper that can be parsed from CLI arguments
+#[derive(Debug, Clone)]
+pub struct MetadataValue(iroha_data_model::Value);
+
+impl FromStr for MetadataValue {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        json5::from_str(s)
+            .map(Self)
```

Comment:
Currently all impls have something like this:
```rust
let smth: some_type = json5::from_str(..).wrap_err(..)?;
Ok(Self(smth))
```
Can we change everything to map(Self) or make it more explicit like in the example so we have a single style?

Reply:
Both options are the same and do the same thing. Both may be used. My option is shorter and straightforward to read in this case.

Reply:
Agree.
Let's then use your solution for other cases?
Otherwise at some point in the future we might add another option for from_str and it'll be very inconvenient.